### ISSUE TITLE
First game bezel is stuck

### ISSUE THREAD

**Issue by stzokev** [thebezelproject/BezelProject-Windows#3]
Just switched from the original bezel_project_web_installer_v1.5.exe which was working excellent for me, but is missing bezels for some games, so I thought I upgrade to the live project. However it doesn't really work. My retroarch installation is a default one, haven't really changed anything (latest 1.7.9). I install the bezels for the chosen platforms (mame, megadrive, nes, snes, psx) and exit the installer. Then I run a game, yupeeee the bezel shows. Then when I choose another game it shows the bezel from the first game I started and it shows it for all irrelevant of platform. I see in retroarch.cfg it has hardcoded the path to the first game e.g: input_overlay = ":\overlays/GameBezels/Megadrive/Adventures of Batman & Robin, The (USA).cfg" If I remove the value, then no bezels are shown for any game. Until i restart the bezel project app where things repeat - first game started shows its bezel for all. 

What am I missing? What is wrong?

**Comment by thebezelproject** [comment]
Not sure what to tell you. Installing on Retroarch 1.7.9 on Windows works great here.

All I can recommend is start from scratch for your Megadrive install related to Retroarch. Remove any core override files under the config directory (note: the project files only supports 'Genesis Plus GX
 and PicoDrive' by default. If you are using a different core, use different names for custom core configs, etc., try setting up with the default Genesis Plux GX core and try that first.

Also check the option in the Windows installer for 'Overwrite existing platform bezel cfg files?' and set this to Yes.

Also make sure your rom names match the files provided with the bezels. If they don't match 100%, you will get a default bezel.

**Comment by stzokev** [comment]
Hi,

Thanks for your reply. I've made sure the config and overlays folders are emptied prior to installing, as the old bezel project installer doesn't have uninstall option. So:

1) There are not custom core configs or core overrides files
2) This happens for ANY platform, not only Megadrive, so while I appreciate your explanation for megadrive supported cores, it is NOT a megadrive specific problem. To clarify this further, if I start Megadrive rom first it would show the bezel for this game. Then if I start Mame or SNES rom, it would still show the bezel from the Megadrive rom. 
3) The rom names are proper, as I mentioned in the original post I am migrating from the old bezel project installer, which works fine and shows correct bezels for all platforms and game
4) I tried to reinstall from scratch today and got this: https://www.screencast.com/t/rZlEZc0zSU Hope that you are aware the installer is being detected as Trojan.

So after all I re-installed the bezel project from scratch and the problem persists. Any other suggestions?

Regards

**Comment by thebezelproject** [comment]
I am aware some Anti-virus apps sometimes detect the app as malware/trojan/virus. The .exe is UPX packed, which is often caught up in the anti-virus alarm report.

I need more info if I am to help. I did what you did today, e.g. installed a fresh download of Retroarch 1.7.9, set the directory paths in Preferences of the app, then installed Megadrive for Retroarch. The bezels show up with running from Retroarch, and Launchbox (the front-end I use).

After you install retroarch and one set of bezels, please provide either a listing, or a pic of the entire retroarch directory with the config and overlays directory expanded so I can see what is inside. Also, attach, or list out the retroarch.cfg file from the main retroarch install directory. Hopefully we can  figure this out.

**Comment by stzokev** [comment]
This is not some Anti-virus app, this is the default Windows 10 antivirus i.e. a lot of people are going to hit that, so it is a bit more serious if its going to affect 99% of your users.

I did what you did, installed only Megadrive and guess what - it works. However as soon as I installed all other platforms I need, the problem reoccurred, so you might want to test my exact use case. I've mentioned all platforms I use in my original post.

If you cannot reproduce this with all platforms installed, then I would provide what you request.

P.S. Just out of curiosity, I installed the old bezel project and everything is fine there as it was.

**Comment by thebezelproject** [comment]
I have installed the other platforms in my Retroarch 1.7.9 install and everything is working as expected. We obviously have a differences in settings somewhere, so please provide all the information you can easily do so we can diagnose further.

**Comment by stzokev** [comment]
Thanks for trying to help. You can find the requested information below:

1) https://drive.google.com/file/d/14Da50XxpK3YQNHLm_4MUP1Qc2Hh8kxYN/view?usp=sharing
2) https://www.screencast.com/t/ETs2kyLEK
3) https://www.screencast.com/t/rLndMM35j and https://www.screencast.com/t/XjrBB1lvrX

Hope these help you track it down.

**Comment by thebezelproject** [comment]
Hehe, everything still runs fine on my end when using your retroarch.cfg file.

OK, next step. Could you archive up your retroarch/config directory and attach it?

**Comment by stzokev** [comment]
Sure, here it is: https://drive.google.com/file/d/1b4j_zU4SILwrcUpPQUTwBAlC87XgHgh4/view?usp=sharing

**Comment by thebezelproject** [comment]
Still no luck finding the problem. Everything just works here using your config directory content.

I guess it is time to step back. Which Retroarch version are you using, the portable or the installer?

Can you provide a step by step that reproduces the problem on your end?

Sorry this is taking so long!

**Comment by stzokev** [comment]
I am using the portable version. Originally started with 1.7.7 and last week I upgraded to 1.7.9 manually as per this video https://www.youtube.com/watch?v=os6txWGIWUQ. As I mentioned earlier and as visible from cfg file, I have not really changed any of the default settings.

**Comment by thebezelproject** [comment]
Portable here too. Hmm, the only thing I can think of at this point is the start from a fresh extract of Retroarch and set the main Retroarch option that automatically saves the configuration to false, then install a bezel pack and see how it goes (this is just a debugging install). I always set this option, but having said that, it should not matter from what I can see. Until I can reproduce the issue here, I can't think of anything else to try.

I assume you are using Windows 10? Where is your Retroarch installation living?

If you can think of anything else to test/try, please let us know.

**Comment by stzokev** [comment]
Yes, Windows 10 ver. 1903 all updates. It resides inside Launchbox, you can see my structure from the screenshot I sent for overlay and config. I would try to start with a fresh install during the weekend and see how it goes.

**Comment by dougsky** [comment]
am I experiencing the same problem ,,, was it solved?

**Comment by thebezelproject** [comment]
Since I can't replicate the error here, I will need more information. If anyone can provide steps for replicating the error, I will investigate further.

**Comment by dougsky** [comment]
https://drive.google.com/drive/folders/1bXHJ04SOvmQHGUBC7oYqxGSGJiA9S_ph

retroarch configuration files are on the gdrive link ... as I can help with information just tell me ... thanks
##################################################################
today I was doing some tests, I did a zero installation of the retroarch, I installed the core of the megadrive genesis_plus_gx_libretro then I had the roms set analyzed. .. then open the first game the Good as in the video ... until and everything is fine, but when opening the second game the bezel applied was from the first game open ... for some games and applied to bezels corresponding to others not. ..

Video: https://youtu.be/d532NiB2-m8

Sorry my english I'm from Brazil and I don't know your language ... I use translator

**Comment by thebezelproject** [comment]
The reason why you are seeing the bezel from the first game when running the second is because you have the Retroarch option to save config on exit set to true. To stop this behavior, exit Retroarch, edit the retroarch.cfg file, search for config_save_on_exit and set the value to "false". Then search for input_overlay and set the value to "".

**Comment by dougsky** [comment]
I just checked the retroarch.cfg file and it already has the settings you mentioned ... the gdrive link I posted contains the retroarch.cfg file if you want to check it out

**Comment by dougsky** [comment]
in the file retroarch.default.cfg it has these definitions below, is there something?

#### Overlay

# Enable the overlay.
# input_overlay_enable = true

# Hide the current overlay from appearing inside the menu.
# input_overlay_hide_in_menu = true

# Path to input overlay.
# input_overlay =

# Opacity of all the UI elements of the overlay.
# input_overlay_opacity = 1.0

# Scale of all UI elements of the overlay.
# input_overlay_scale = 1.0

**Comment by thebezelproject** [comment]
No, the retroarch.default.cfg isn't used once you start Retroarch the first time and a retroarch.cfg is created.

Since the settings in retroarch.cfg are correct, please check any core specific cfg files under the config directory and make sure those settings are set correctly in those files too.

**Comment by dougsky** [comment]
I checked all files and the command lines are the same as the panels that work .... very strange such a bug ... The names of the roms influence the operation ... you would have to pass me the pack of roms that you use so that can i test?

**Comment by dougsky** [comment]
I did another test from scratch at that moment ... and I started opening game by game ..... many of them the bezels were not applied and in others the panels were applied correctly ... I think it has to do with some information from roms. Example in game 3 Ninjas Kick Back (USA) that is the first of the list it runs without the bezels ... for today it arrives ... until !!!

**Comment by thebezelproject** [comment]
If you no longer have the wrong bezel being displayed when you change games, that is progress. If some games show the correct bezel, but other games that don't show any bezel, that shows you have a naming conflict between your games and the cfg files under the retroarch\config directory. You can either rename the games, or the cfg files. Whatever works best for you.

**Comment by dougsky** [comment]
Hello grandmaster, I found out where the error is, I'll take as an example the game 3 Ninjas Kick Back that I mentioned yesterday, following the rom file has the following name 3 Ninjas Kick Back.zip and the configuration file that makes reference to the bezel has the name 3 Ninjas Kick Back (USA) .cfg ... I unzipped the rom and saw that her name was like 3 Ninjas Kick Back.MD and then unzipped another rom that was sure that the panel was applied and the name of that unzipped rom the file is with the name of the game (USA) .MD I took the rom of 3 Ninjas Kick Back and added (USA) to the end of the name and had it compressed again and to my surprise ... the panel was applied, so the name and the internal compressed file.

**Comment by dougsky** [comment]
I made an illustration to better exemplify
https://drive.google.com/file/d/1Ejy-kxkFZhZmr47WwL6x__Da918hKWGD/view

**Comment by stzokev** [comment]
Good for you, but this was not my problem. My files were named correctly and as I pointed out originally, they work with the old bezel project (which I am currently using). It is only this bezel project that I am having problems with. Since I do not anticipate many new games would come up for these old consoles/arcades, I would just stick to the old project as it works for me.

**Comment by dougsky** [comment]
@stzokev As I mentioned, I went through the same bug as you, see the video attached to the post, error is relating to the internal names of the file of the roms ...
